compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -6400,6 +6400,10 @@ public boolean isOrderedType(BType type, boolean hasCycle) {
                 boolean allMembersOrdered = false;
                 BType firstTypeInUnion = getTypeWithEffectiveIntersectionTypes(getReferredType(
                         memberTypes.stream().filter(m -> !isNil(m)).findFirst().orElse(memberTypes.iterator().next())));
+                if (isNil(firstTypeInUnion)) {
+                    // Union contains only the nil type.
```

Comment:
Is this comment guaranteed that the union has only nil type? I was expecting to cache the result in a boolean and use inside for loop.

Reply:
yes you are correct..